server/Schemas/resolvers.js

```diff
@@ -40,11 +41,12 @@ const resolvers = {
           return { token, user };
         },
         saveBook: async (parent, args, context) => {
-            if (context.user) {
+            const user = await authenticate(context.auth);
```

Comment:
To use the authenticate function, pass the authorization token to the authenticate function.